neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/impl/BoltNeo4jConnectionImpl.java

```diff
@@ -316,15 +314,23 @@ public void closeSession(Session session){
 			return false;
 		}
 
-		Runnable r = new Runnable() {
-			@Override public void run() {
-				Session s = getSession();
-				Transaction tr = getTransaction();
+		Runnable r = () -> {
+			Session s = newNeo4jSession();
+			Transaction tr = s.beginTransaction();
+			try {
```

Comment:
this block is really odd, shouldn't it use try-with-resources
for both session and transaction?

Reply:
Yes it's odd I improved it